asmcomp/arm/emit.mlp

```diff
@@ -798,26 +798,48 @@ let emit_instr i =
 
 (* Emission of an instruction sequence *)
 
+(* consider the degenerate case where a single literal is followed by
+   a jump table longer than 4KB; we have to emit the constant pool
+   before the jump table or it will be too late *)
+let emit_instr_before_pool i =
+  match i.desc with
+  | Lswitch jumptbl -> 2 + Array.length jumptbl
+  | _ -> emit_instr i
```

Comment:
> Then instead of changing fallthrough, which seems a bit misleading, you could just test for Some and avoid the third match; you just emit the instruction in the Some. It might not be worth having emit_instr_after_pool at that point.

I don't understand what you are saying here (quite ironic given that you evidently consider it so obvious that you've used "just" twice).